src/Core/src/Platform/Android/ApplicationExtensions.cs

```diff
@@ -45,6 +45,17 @@ public static void CreatePlatformWindow(this Activity activity, IApplication app
 
 			var window = application.CreateWindow(activationState);
 
+			if (window.Handler?.PlatformView is Activity oldActivity && 
+				oldActivity != activity &&
+				!oldActivity.IsDestroyed)
+			{
+				throw new InvalidOperationException(
+					$"This window is already associated with an active Activity ({oldActivity.GetType()}). " + 
+					$"Please override CreateWindow on {application.GetType()} "  + 
+					$"to add support for multiple activities https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/windows?view=net-maui-8.0#create-a-window "  + 
+					$"or set the LaunchMode to SingleTop on {activity.GetType()}.");
+			}
```

Comment:
Is there a way to write a test for this that checks the exception?

Reply:
Wrote a test